test/framework/src/main/java/org/elasticsearch/test/client/RandomizingClient.java

```diff
@@ -84,4 +87,18 @@ public Client in() {
         return super.in();
     }
 
+    private class RandomizedSearchRequestBuilder extends SearchRequestBuilder {
+        RandomizedSearchRequestBuilder(ElasticsearchClient client) {
+            super(client);
+        }
+
+        @Override
+        public SearchRequestBuilder setPointInTime(PointInTimeBuilder pointInTimeBuilder) {
+            if (defaultPreference != null) {
+                setPreference(null);
+            }
+            return super.setPointInTime(pointInTimeBuilder);
```

Comment:
I guess we should also override setPreference to make sure it throws if point in time is already set, otherwise we rely on order in which the methods are called?

Reply:
I am just checking the default preference here since it is set by the `RandomizingClient` ctr. If a test using this client sets the preference explicitly an error will be thrown during the execution indicating to the author of the test that preference shouldn't be used in this context.

Reply:
oh right!